### PR TITLE
updated plugin_shoutbox to v1.02.4

### PR DESCRIPTION
**Pull request by surrim**
* Use serendipity_db_escape_string and avoid sprintf

### CODE REVIEW

File: serendipity_plugin_shoutbox/serendipity_plugin_shoutbox.php [hunk 1]
```diff
@@ -198,19 +188,19 @@ function generate_content(&$title)
...
+            ORDER BY timestamp DESC
+               LIMIT $max_entries";
 ?>
 <div class="serendipity_shoutbox">
 <?php
         $sql = serendipity_db_query($q);
         if ($sql && is_array($sql)) {
             foreach($sql AS $key => $row) {
-                $comments = wordwrap(strip_tags($row['comment']), $max_chars, '@@@', 1);
+                $comments = serendipity_specialchars(wordwrap($row['comment'], $max_chars, '@@@', 1));
```
Comment (onli) [inline serendipity_plugin_shoutbox/serendipity_plugin_shoutbox.php:203]: Just a note: `serendipity_specialchars` was introduced to fix a regression/design issue with charset support in PHP 5.4 and 5.5. It's something we could remove by now -> it is okay to call the PHP `htmlspecialchars` directly, if prefered,

File: serendipity_plugin_shoutbox/serendipity_plugin_shoutbox.php [hunk 2]
```diff
@@ -227,7 +217,7 @@ function generate_content(&$title)
                 serendipity_plugin_api::hook_event('frontend_display', $entry);
                 $entry['comment'] = wordwrap($entry['comment'], $wordwrap, "\n", 1);
 
-                echo '<div class="serendipity_shoutbox_date">' . (function_exists('serendipity_specialchars') ? serendipity_specialchars(serendipity_strftime($dateformat, $row['stamp'])) : htmlspecialchars(serendipity_strftime($dateformat, $row['stamp']), ENT_COMPAT, LANG_CHARSET)) . '</div>' . "\n"
+                echo '<div class="serendipity_shoutbox_date">' . serendipity_specialchars(serendipity_strftime($dateformat, $row['stamp'])) . '</div>' . "\n"
```
Comment (onli) [inline serendipity_plugin_shoutbox/serendipity_plugin_shoutbox.php:220]: With this change (and the unguarded function call above) the minimal serendipity function should be set to when that function was introduced, which was `2.0`.

**Comment by surrim** [comment]
I updated the branch. Sorry for using the old function. Would be nice to have this information in the docs ;-)

**Comment by onli** [comment]
No worries, it would have been consistent and there was no real decision on this - didn't come up for a while. I'll merge now, thanks for the PR and the applied changes :)

**Pull request merged**